### PR TITLE
Formula Editor Refactoring

### PR DESCRIPTION
**Pull request by obusher**
* Remove of Log info output.
* Resolving of warnings in formulaeditor classes.
* Introducing some new util methods for testing.

### CODE REVIEW

File: catroid/src/org/catrobat/catroid/formulaeditor/Formula.java [hunk 1]
```diff
@@ -232,7 +227,7 @@ public boolean isSingleNumberFormula() {
 	}
 
 	@Override
-	public Formula clone() {
+	public Formula clone()  {
```
Comment (robertpainsi) [inline catroid/src/org/catrobat/catroid/formulaeditor/Formula.java]: Remove second whitespace

**Comment by robertpainsi** [comment]
This is awesome!
It looks like you improved the code by using all the checkstyle modules which aren't integrated yet. (https://jira.catrob.at/browse/CAT-1254)

Are these all warnings by the IDE or how did you do that?

**Comment by obusher** [comment]
Yes this are the warnings from the IDE. :8ball:

File: catroid/src/org/catrobat/catroid/formulaeditor/InternFormulaParser.java [hunk 2]
```diff
@@ -145,12 +144,11 @@ public FormulaElement parseFormula() {
 		try {
 			List<InternToken> copyIternTokensToParse = new ArrayList<InternToken>(internTokensToParse);
 			if (InternFormulaUtils.applyBracketCorrection(copyIternTokensToParse)) {
-				Log.i("info", "applyBracketCorrection-> TRUE");
 				internTokensToParse.clear();
 				internTokensToParse.addAll(copyIternTokensToParse);
 			}
 		} catch (EmptyStackException emptyStackException) {
-			Log.i("info", "emptyStackException-> TRUE");
+			Log.d(getClass().getSimpleName(), "Bracket correction failed.", emptyStackException);
```
Comment (robertpainsi) [inline catroid/src/org/catrobat/catroid/formulaeditor/InternFormulaParser.java]: Please use a Tag instead like here:
https://github.com/Catrobat/Catroid/blob/develop/catroid/src/org/catrobat/catroid/ProjectManager.java#L66

File: catroid/src/org/catrobat/catroid/formulaeditor/Operators.java [hunk 3]
```diff
@@ -25,9 +25,9 @@
...
 
 public enum Operators {
-	LOGICAL_AND(2, true), LOGICAL_OR(1, true), EQUAL(3, true), NOT_EQUAL(4, true), SMALLER_OR_EQUAL(4, true), GREATER_OR_EQUAL(
-			4, true), SMALLER_THAN(4, true), GREATER_THAN(4, true), PLUS(5), MINUS(5), MULT(6), DIVIDE(6), MOD(6), POW(
-			7), LOGICAL_NOT(4, true);
+	LOGICAL_AND(2, true), LOGICAL_OR(1, true), EQUAL(3, true), NOT_EQUAL(4, true), SMALLER_OR_EQUAL(4, true),
+	GREATER_OR_EQUAL(4, true), SMALLER_THAN(4, true), GREATER_THAN(4, true), PLUS(5), MINUS(5), MULT(6), DIVIDE(6),
+	MOD(6), POW(7), LOGICAL_NOT(4, true);
 
 	private static final String TAG = Operators.class.getSimpleName();
```
Comment (robertpainsi) [inline catroid/src/org/catrobat/catroid/formulaeditor/Operators.java]: Unfortunately  I can't add a comment on the next line...
```private final Integer priority;``` Would a primitive int be sufficient too?

File: catroid/src/org/catrobat/catroid/ui/fragment/FormulaEditorListFragment.java [hunk 4]
```diff
@@ -184,21 +180,17 @@ public void showFragment(Context context) {
 
 	@Override
 	public boolean onKey(DialogInterface d, int keyCode, KeyEvent event) {
-		Log.i("info", "onKey() in FE-ListFragment! keyCode: " + keyCode);
-		boolean returnValue = false;
 		switch (keyCode) {
```
Comment (robertpainsi) [inline catroid/src/org/catrobat/catroid/ui/fragment/FormulaEditorListFragment.java]: Is this a switch with only a single case? Maybe an if would be better?

**Comment by robertpainsi** [comment]
All of the changes look fine!
I didn't inspect the test changes very well but mostly they are refactoring stuff. Maybe there are some new testcases too. However, if the testrun is ok, I will merge it.

**Comment by robertpainsi** [comment]
Thanks for the info @obusher 
I like the Android Studio warnings and your improvements a lot!!! :heart_eyes_cat:

File: catroid/src/org/catrobat/catroid/ui/fragment/FormulaEditorFragment.java [hunk 5]
```diff
@@ -120,9 +119,7 @@ private void resetActionBar() {
 	}
 
 	public static void showFragment(View view, Brick brick, Formula formula) {
-
-		SherlockFragmentActivity activity = null;
-		activity = (SherlockFragmentActivity) view.getContext();
+		SherlockFragmentActivity activity  = (SherlockFragmentActivity) view.getContext();
```
Comment (robertpainsi) [inline catroid/src/org/catrobat/catroid/ui/fragment/FormulaEditorFragment.java]: 2 spaces :scream_cat:

File: catroidTest/src/org/catrobat/catroid/test/formulaeditor/ParserTestStringFunctions.java [hunk 6]
```diff
@@ -65,136 +64,131 @@ protected void setUp() {
...
 		index = "3";
-		FormulaEditorUtil.testDoubleParameterFunction(Functions.LETTER, InternTokenType.NUMBER, index,
+		FormulaEditorTestUtil.testDoubleParameterFunction(Functions.LETTER, InternTokenType.NUMBER, index,
 				InternTokenType.USER_VARIABLE, PROJECT_USER_VARIABLE_NAME2,
 				String.valueOf(USER_VARIABLE_2_VALUE_TYPE_STRING.charAt(Integer.valueOf(index) - 1)), testSprite);
 
-		List<InternToken> firstParameterList = new LinkedList<InternToken>();
-		List<InternToken> secondParameterList = new LinkedList<InternToken>();
-		firstParameterList = FormulaEditorUtil.buildBinaryOperator(InternTokenType.NUMBER, "5", Operators.PLUS,
+		List<InternToken> firstParameterList  = FormulaEditorTestUtil.buildBinaryOperator(InternTokenType.NUMBER, "5", Operators.PLUS,
```
Comment (robertpainsi) [inline catroidTest/src/org/catrobat/catroid/test/formulaeditor/ParserTestStringFunctions.java]: 2 spaces :scream_cat:

**Comment by obusher** [comment]
Yea it its only refactoring, otherwise it would be confusing.

**Pull request merged**